### PR TITLE
Fix reference to server vs. .com

### PR DESCRIPTION
**Pull request by rrhyne**
This pull request fixes 'Managed Sourcegraph instance' being used to describe Sourcegraph server and ' Sourcegraph server' defined as Sourcegraph.com.

### CODE REVIEW

File: handbook/communication/content_guidelines/style_and_mechanics.md [hunk 1]
```diff
@@ -673,8 +673,8 @@ We describe ourselves with a few different names depending on context, and we sh
 
 - **Sourcegraph**: Main product. Prefer using this name unless you need to be more precise.
 - **Self-hosted Sourcegraph instance**: Only if clarification between Sourcegraph Cloud, managed instances, and on-premises instances is required.
-- **Managed Sourcegraph instance**: Only if clarification between the Sourcegraph Cloud instance, self-hosted instances, and managed instances is required.
-- **Sourcegraph Server**: The public instance of Sourcegraph for open source code at [sourcegraph.com](https://sourcegraph.com).
+- **Sourcegraph Server**: Only if clarification between the Sourcegraph Cloud instance, self-hosted instances, and managed instances is required.
+- **Sourcegraph.com**: The public instance of Sourcegraph for open source code at [sourcegraph.com](https://sourcegraph.com).
```
Comment (christinaforney) [inline handbook/communication/content_guidelines/style_and_mechanics.md:677]:
```suggestion
- **Sourcegraph Cloud**: The public instance of Sourcegraph for open source code at [sourcegraph.com](https://sourcegraph.com).
```

File: handbook/communication/content_guidelines/style_and_mechanics.md [hunk 2]
```diff
@@ -673,8 +673,8 @@ We describe ourselves with a few different names depending on context, and we sh
 
 - **Sourcegraph**: Main product. Prefer using this name unless you need to be more precise.
 - **Self-hosted Sourcegraph instance**: Only if clarification between Sourcegraph Cloud, managed instances, and on-premises instances is required.
```
Comment (christinaforney) [inline handbook/communication/content_guidelines/style_and_mechanics.md:675]:
```suggestion
- **Self-hosted Sourcegraph instance or Sourcegraph Server**: The self-hosted Sourcegraph instance maintained by customers. Only if clarification between Sourcegraph Cloud, managed instances, and on-premises instances is required.
```

File: handbook/communication/content_guidelines/style_and_mechanics.md [hunk 3]
```diff
@@ -673,8 +673,8 @@ We describe ourselves with a few different names depending on context, and we sh
...
-- **Managed Sourcegraph instance**: Only if clarification between the Sourcegraph Cloud instance, self-hosted instances, and managed instances is required.
-- **Sourcegraph Server**: The public instance of Sourcegraph for open source code at [sourcegraph.com](https://sourcegraph.com).
+- **Sourcegraph Server**: Only if clarification between the Sourcegraph Cloud instance, self-hosted instances, and managed instances is required.
```
Comment (christinaforney) [inline handbook/communication/content_guidelines/style_and_mechanics.md:676]:
```suggestion
- **Managed Sourcegraph instance**: A fully isolated Sourcegraph Server deployment maintained by the Sourcegraph team for a customer. Only use if clarification between the Sourcegraph Cloud instance, self-hosted instances, and managed instances is required.
```

**Review by christinaforney — CHANGES_REQUESTED** [review]
Added some additional corrections.

**Comment by rrhyne** [comment]
Ah, I missed all that. Thanks.

**Review by christinaforney — APPROVED** [review]

**Pull request merged**